pallets/nft/src/lib.rs

```diff
@@ -59,6 +60,13 @@ pub trait Trait<I = DefaultInstance>: system::Trait {
     type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
 }
 
+// The asset type and the creation time.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, Default, RuntimeDebug)]
+pub struct AssetData<AssetInfo, Moment> {
+    pub info: AssetInfo,
+    pub creation: Moment
```

Comment:
What is the reasoning behind hard-coding this into the NFT pallet and not allowing NFT implementers the freedom to include (or exclude) this information from their custom `AssetInfo` type?

Reply:
The timestamp adds a type of unique identifier that will prove the nft can't be replaced in the future even if you burn it and then recreate a new one with the same id.

Reply:
I feel like this may be an implementation detail I would prefer to leave up to the asset admin. I think this capability is already exposed by way of a generic `AssetInfo` interface and I don't really think I want to push this logic down into a generic NFT pallet.